src/mono/System.Private.CoreLib/src/System/Collections/Generic/EqualityComparer.Mono.cs

```diff
@@ -39,6 +39,11 @@ private static EqualityComparer<T> CreateComparer()
             {
                 return (EqualityComparer<T>)(object)(new ByteEqualityComparer());
             }
+            else if (t == typeof(string))
```

Comment:
There is a comment on line 34 about "METHOD-TO-IR.C should be kept in sync. Is that still valid?